### PR TITLE
Hot Fix for Custom Browser Options (#244)

### PR DESCRIPTION
**Pull request by MahmoudElSharkawy**
This is a hot fix for Custom Browser Options for the following ticket (#244) to enable adding browser option(s) while still using the default options of SHAFT.
As this is a hot fix, there is a known issue that when adding option(s) that's already included in the default options of SHAFT, the SHAFT default option(s) will override the custom one(s) of the same type.

### CODE REVIEW

**Comment by MahmoudElSharkawy** [comment]
Added a feature which the user can choose to use/append to the default browsers options or not by adding a boolean appendToDefaultOptions parameter to the getBrowser() methods and updated the setDriverOptions() method accordingly.

**Comment by ShorouqElrakhawy** [comment]
@MohabMohie May I know the available time for this PR to be merged?

File: src/main/java/com/shaft/gui/browser/BrowserFactory.java [hunk 1]
```diff
@@ -317,16 +321,19 @@ private static void setLoggingPrefrences() {
         logPrefs.enable(LogType.DRIVER, Level.ALL);
     }
 
-    private static void setDriverOptions(String browserName, MutableCapabilities customBrowserOptions) {
+    private static void setDriverOptions(String browserName, MutableCapabilities customBrowserOptions, Boolean appendToDefault) {
```
Comment (MohabMohie) [inline src/main/java/com/shaft/gui/browser/BrowserFactory.java]: I think this boolean variable is redundant, I'll share my detailed feedback offline

**Comment by MohabMohie** [comment]
@ShorouqElrakhawy it's still ongoing, please hold.

**Review by MohabMohie — APPROVED** [review]

**Comment by MohabMohie** [comment]
Merged with known issue:
Ordering for options is reversed due to bug in Selenium WebDriver options.merge() method which drops a lot if the settings.

For now if you define a custom property it will be respected.
But if it was already set by the default engine configuration it will be ignored and the default will supercede.

**Comment by MohabMohie** [comment]
@MahmoudElSharkawy please follow this commit for the bug fix from Selenium WebDriver side

https://github.com/SeleniumHQ/selenium/commit/ec9c67b2fb1ec110f51934832fec3cef8a9bf746

**Pull request merged**